<?xml version='1.0' encoding='utf-8'?>
<robot name="scout_urdf" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:arg name="prefix" default="scout_" />
  <xacro:arg name="mode" default="simulation" />
  <xacro:arg name="base_name" default="base" />
  <xacro:arg name="ros2_control_config_urdf_file" default="" />
  <xacro:arg name="controller_manager_config_yaml_file" default="" />
  <xacro:arg name="ros_prefix" default="/scout/" />

  <xacro:include filename="$(find scout_description)/urdf/scout.xacro" />
  <xacro:scout model="v2" mode="$(arg mode)" prefix="$(arg prefix)" />

  <gazebo>
    <plugin filename="libromea_gazebo_ros2_control.so" name="gazebo_ros2_control">
      <ros2_control_config_file>$(arg ros2_control_config_urdf_file)</ros2_control_config_file>
      <controller_manager_config_file>$(arg controller_manager_config_yaml_file)</controller_manager_config_file>
      <ros>
        <namespace>$(arg ros_prefix)$(arg base_name)</namespace>
      </ros>
    </plugin>
  </gazebo>

</robot>